### DISCUSSION TITLE
Tests with async imports randomly fail

### DISCUSSION THREAD

**Discussion by freakzlike** [vitest-dev/vitest#2577] [Q&A] [answered]
Hi,
I'm facing an issue where tests with async imports randomly fail, but only on our CI server. My vue project has about ~2200 tests and I run `await vi.dynamicImportSettled()` on every required test. I don't have any issue on my local PC or within local CI docker image. Currently I run two separate CI jobs (on same host) which split all tests with `--shard=1/2` and the issue does not occur either.

I have the suspicion that this is somehow a timing issue as the CI server might be too slow.

Maybe someone as an idea...
Thanks in advance

**Comment by sheremet-va** [comment, marked as answer]
Should be fixed in the next release. dynamicImportSettled wasn't reliable enough

**Reply by freakzlike** [reply]
Working with v0.26.3 and also seems a bit faster? Thank you 👍
